cmd/instance_show.go

```diff
@@ -93,106 +106,139 @@ func (c *instanceShowCmd) cmdRun(cmd *cobra.Command, _ []string) error {
+		out.DeployTarget = dt.Name
 	}
 
-	if instance.ElasticIPIDs != nil {
-		for _, id := range *instance.ElasticIPIDs {
-			elasticIP, err := globalstate.EgoscaleClient.GetElasticIP(ctx, c.Zone, id)
+	if instance.ElasticIPS != nil {
+		for _, eip := range instance.ElasticIPS {
+			resp, err := client.ListElasticIPS(ctx)
+			if err != nil {
+				return err
+			}
+			found_eip, err := resp.FindElasticIP(eip.ID.String())
 			if err != nil {
 				return fmt.Errorf("error retrieving Elastic IP: %w", err)
 			}
-			out.ElasticIPs = append(out.ElasticIPs, elasticIP.IPAddress.String())
+			out.ElasticIPs = append(out.ElasticIPs, found_eip.IP)
 		}
 	}
 
-	instanceType, err := globalstate.EgoscaleClient.GetInstanceType(ctx, c.Zone, *instance.InstanceTypeID)
+	it, err := client.GetInstanceType(ctx, instance.InstanceType.ID)
 	if err != nil {
 		return err
 	}
-	out.InstanceType = fmt.Sprintf("%s.%s", *instanceType.Family, *instanceType.Size)
+	out.InstanceType = fmt.Sprintf("%s.%s", it.Family, it.Size)
 
-	if instance.PrivateNetworkIDs != nil {
-		for _, id := range *instance.PrivateNetworkIDs {
-			privateNetwork, err := globalstate.EgoscaleClient.GetPrivateNetwork(ctx, c.Zone, id)
+	if instance.PrivateNetworks != nil {
+		for _, pn := range instance.PrivateNetworks {
+			resp, err := client.ListPrivateNetworks(ctx)
+			if err != nil {
+				return err
+			}
+			found_pn, err := resp.FindPrivateNetwork(pn.ID.String())
 			if err != nil {
 				return fmt.Errorf("error retrieving Private Network: %w", err)
 			}
-			out.PrivateNetworks = append(out.PrivateNetworks, *privateNetwork.Name)
+			out.PrivateNetworks = append(out.PrivateNetworks, found_pn.Name)
 		}
 	}
 
-	if instance.SecurityGroupIDs != nil {
-		for _, id := range *instance.SecurityGroupIDs {
-			securityGroup, err := globalstate.EgoscaleClient.GetSecurityGroup(ctx, c.Zone, id)
+	if instance.SecurityGroups != nil {
+		for _, sg := range instance.SecurityGroups {
+			resp, err := client.ListSecurityGroups(ctx)
+			if err != nil {
+				return err
+			}
+			found_sg, err := resp.FindSecurityGroup(sg.ID.String())
 			if err != nil {
 				return fmt.Errorf("error retrieving Security Group: %w", err)
 			}
-			out.SecurityGroups = append(out.SecurityGroups, *securityGroup.Name)
+			out.SecurityGroups = append(out.SecurityGroups, found_sg.Name)
 		}
 	}
 
-	template, err := globalstate.EgoscaleClient.GetTemplate(ctx, c.Zone, *instance.TemplateID)
+	template, err := client.GetTemplate(ctx, instance.Template.ID)
 	if err != nil {
 		return err
 	}
-	out.Template = *template.Name
+	out.Template = template.Name
 
-	rdns, err := globalstate.EgoscaleClient.GetInstanceReverseDNS(ctx, c.Zone, *instance.ID)
+	rdns, err := client.GetReverseDNSInstance(ctx, instance.ID)
 	if err != nil {
-		if errors.Is(err, exoapi.ErrNotFound) {
+		if errors.Is(err, v3.ErrNotFound) {
 			out.ReverseDNS = ""
 		} else {
 			return err
 		}
+	} else if rdns == nil {
+		out.ReverseDNS = ""
+	} else {
```

Comment:
Unsure if that check is needed, rdns being nil should be covered by 
```
if errors.Is(err, v3.ErrNotFound) {
			out.ReverseDNS = ""
		}
```
just above